src/Components/Post/index.js

```diff
@@ -46,7 +47,10 @@ function Post() {
 
   useEffect(() => {
     setComments(post.comments?.data || [])
-  }, [post])
+  }, [loading])
```

Comment:
You create another dependency issue, instead of fixing the current one. 
What if `post` will be undefined (`setComments` line)?